### ISSUE TITLE
Few questions while reading your paper

### ISSUE THREAD

**Issue by kzhoa** [rk2900/DLF#1]
Hi Ren,

I have read your paper and encountered some questions.

## 1.What's the winning condition in one auction?

it confused me that your paper gives a winning condition as z<b,
![image](https://user-images.githubusercontent.com/6019006/58218980-aec1f500-7d3b-11e9-8377-a4fbf5b7438d.png)

while you give uncensored condition $z \le b$ in README.

![image](https://user-images.githubusercontent.com/6019006/58219077-23952f00-7d3c-11e9-9b04-83ba992a0773.png)

and I also found an interesting statement about this dataset in a  [previous published paper](https://arxiv.org/abs/1407.7073)
<br />
![image](https://user-images.githubusercontent.com/6019006/58219197-8b4b7a00-7d3c-11e9-9e2c-c8b25dd7b58a.png)

so could an advertiser win this auction with z equals b ?

I assume not for now.
<br />

## 2.What does h denotes in your paper ?
<br />

the defination in the paper is attached here
<br />
![image](https://user-images.githubusercontent.com/6019006/58219491-bda9a700-7d3d-11e9-9362-40659a3fb4bd.png)

so let's take h as 'the just winning probability given z>=b_{l-1}'.
but if  z  is in V_{l} = (bl ,bl+1] , it means z>bl , then how could we win by b_{l}?

since z need to be at least no greater than b could the advertiser win, I will give my understanding of  h
<br />
![image](https://user-images.githubusercontent.com/6019006/58219667-5a6c4480-7d3e-11e9-8ad1-109a7024d984.png)

and if you would confirm that one cannot win with z==b , I believe the defination of V_{l} should be [bl,bl+1). 

Only in this way can you get the equation h_{l} = Pr(z \in V_{l-1} | z \ge b_{l-1}).
![image](https://user-images.githubusercontent.com/6019006/58220750-ac16ce00-7d42-11e9-934a-257eecc02ac7.png)

## 3.What's the exact meaning of each h produced by RNN Cell?

after the second question, I'm trapped by a even more bigger one reading your code.

Assume that we still have h as **the just winning probability given z>=b_{l-1}**, which should be consistent both in your paper and code.

by the code below you prod all 'h's tegother , which is unreasonble.
```
survival_rate_last_one = tf.reduce_prod(x[0:bid_len])
anlp_rate_last_one = tf.reduce_prod(x[0:market_len + 1])
anlp_rate_last_two = tf.reduce_prod(x[0:market_len])
```

this enlights me that output of RNN maynot be the 'h' in your paper.
instead it looks like **the losing probability at b_{l} given z>=b_{l-1}** ,not winning , i.e. **the survival rate at b_{l}**

so I wonder whether you have given a reverse defination in the paper about h?
if not, what's the output of RNN stand for?

thanks

**Comment by rk2900** [comment]
Thank you for your comments.
1. Only the samples with `b>z` are the winning cases.
In our code, line 59 in BASE_MODEL.py, we define that `bid_price <= market_price` is the losing condition, thus the sample satisfying that would be added into the subsequent procedure for the losing cases.
I've fix the typo in README.md in our repo. Only those samples with 'b>z' is the winning cases.

2. Here "h" means "z in the winning interval while not in the winning interval over the small price range". It might be confusing while reading the paper. But what I mean is that, "z = b-1 while z >= b-1".
You may understand this statement by assuming "b_l - b_{l-1} = 1".

3. We implement DLF model with some tricks, but it's correct.
The cell at each timestep in our model predicts "h" with binary probability output through sigmoid function, which means the cell at each timestep will output "h" and "1-h". In our paper we need to multiply these "1-h" from the first timestep to the last timestep (from 1 to b), to estimate "S(b)". Thus we just choose one of the binary outputs, to do the subsequent inference. We do not need to care the other output in the subsequent calculation.
And p_z = S(b) - S(b+1) and has been implemented as that in line 320 of BASE_MODEL.py, which will be used for ANLP calculation.
So our implementation is correct and we just get the output to represent (1-h).
It is all right to choose another output from the binary result of the cell at each timestep. It's just an implementation trick to simplify calculation.

**Comment by kzhoa** [comment]
Hey! Thanks so much for the detailed response! I have figured these out and  make run successfully!
Really appreciated of your work and open sourcing code!
